drivers/sensor/amg88xx/amg88xx_trigger.c

```diff
@@ -165,24 +163,26 @@ int amg88xx_init_interrupt(const struct device *dev)
 	struct amg88xx_data *drv_data = dev->data;
 	const struct amg88xx_config *config = dev->config;
 
-	/* setup gpio interrupt */
-	drv_data->gpio = device_get_binding(config->gpio_name);
-	if (drv_data->gpio == NULL) {
-		LOG_DBG("Failed to get pointer to %s device!",
-			config->gpio_name);
-		return -EINVAL;
-	}
+	drv_data->cfg = config;
+
+	if (!device_is_ready(config->int_gpio.port)) {
+		if (config->int_gpio.port) {
+			LOG_ERR("%s: device %s is not ready", dev->name,
+						config->int_gpio.port->name);
+			return -ENODEV;
+		}
 
-	drv_data->gpio_pin = config->gpio_pin;
+		LOG_DBG("%s: int_gpio not defined in DT", dev->name);
+		return 0;
+	}
```

Comment:
Driver uses `GPIO_DT_SPEC_INST_GET`, not `GPIO_DT_SPEC_INST_GET_OR`, meaning `int_gpio.port` can't be NULL. We should also return `-ENODEV`.